From f68502f22358620914a49be3af583cec58458da0 Mon Sep 17 00:00:00 2001
From: William Jordan <bjordan@habana.ai>
Date: Wed, 07 May 2025 21:54:30 +0300
Subject: [PATCH] [SW-223216] fix fw_info alignment

fw_info structure shouldn't use type enum;
the size of the enum is not complier independent.
Changed the element to uint32_t.

Change-Id: Ida9dcc04191da822957d9c68b183bf962743d234
Signed-off-by: William Jordan <bjordan@habana.ai>
---

diff --git a/hl_fw_if.h b/hl_fw_if.h
index 679596e..c7a5a9a 100644
--- a/hl_fw_if.h
+++ b/hl_fw_if.h
@@ -130,7 +130,9 @@
 	uint32_t cpu_boot_err_val[2];	/* boot error registers for passing between components */
 	uint32_t fw_rsvd_mem_mb;	/* reserved DRAM memory size in MB for FW/SVE use */
 	uint32_t cpu_boot_sts_val[2];	/* boot status registers for passing between components */
-	enum hbm_mem_scrub_status mem_scrub_status;
+	uint8_t mem_scrub_status;	/* enum hbm_mem_scrub_status */
+	uint8_t reserved1;
+	uint16_t reserved2;
 	uint32_t serdes_lfs_partition_offset;	/*The offset in flash of serdes data partition */
 	uint32_t serdes_lfs_partition_size;	/*The size of serdes data partition */
 } __packed;
